src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -689,24 +694,26 @@ public virtual HtmlString GenerateAntiforgery([NotNull] ViewContext viewContext)
 
                     if (!string.IsNullOrEmpty(errorText))
                     {
-                        var listItem = new TagBuilder("li", _htmlEncoder);
+                        var listItem = new TagBuilder("li");
                         listItem.SetInnerText(errorText);
-                        htmlSummary.AppendLine(listItem.ToString(TagRenderMode.Normal));
+                        htmlSummary.Append(listItem.ToHtmlContent(TagRenderMode.Normal));
+                        htmlSummary.Append(Environment.NewLine);
                     }
                 }
             }
 
-            if (htmlSummary.Length == 0)
+            if (htmlSummary.ToString().Length == 0)
```

Comment:
Actually, I could work around this by setting a flag when `htmlSummary.Append()` is called with the `listItem`.